### PR TITLE
Migrating  bytebuffercollections from Metamarkets.

### PR DESCRIPTION
**Pull request by akashdw**
Migrating bytebuffercollections from metamarkets. This migration will be helpful to deal problems around having separate PRs and release cycle of Druid, extendedset and bytebuffercollections. This PR is continuation of https://github.com/druid-io/druid/pull/3585. 

Note: External extensions might have to change.

### CODE REVIEW

**Comment by cheddar** [comment]
Please note the differing license headers on `extendedset` and `bytebuffercollections` files.  `extendedset` is actually a fork of someone else's repository, given that it is a fork already, the statement that it was "licensed to Metamarkets under a CLA" is not true, so we instead opted to keep the same license header.

**Comment by drcrallen** [comment]
@cheddar given that, can we split the "purely MMX CLA" into its own PR for easier merging?

I'm a bit concerned about how we can make sure we can maintain a proper licensing header for things that did not originate under a MMX CLA

**Comment by akashdw** [comment]
@drcrallen @cheddar Removed extendedset module.

**Comment by fjy** [comment]
@akashdw there are some code conflicts

**Comment by cheddar** [comment]
@fjy the code conflicts will happen on almost every PR merge because imports.  It would be great if we could get signoff on the general idea from @drcrallen or @leventov as representatives of MMX (given that the code is coming from an MMX repo) before resolving the conflicts.  Otherwise, there will be constant chasing of conflicts.

**Comment by leventov** [comment]
@cheddar I leave this to @drcrallen.

File: bytebuffer-collections/src/main/java/io/druid/collections/bitmap/MutableBitmap.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+{
+  /**
+   * Empties the content of this bitmap.
+   */
+  public void clear();
+
+  /**
+   * Compute the bitwise-or of this bitmap with another bitmap. The current
+   * bitmap is modified whereas the other bitmap is left intact.
+   * <p>
```
Comment (cheddar) [inline bytebuffer-collections/src/main/java/io/druid/collections/bitmap/MutableBitmap.java]: These dangling `<p>`s apparently get in the way of the building of javadoc for Druid.  They should be cleaned up.  It should be acceptable to just replace `<p>` with a space.

**Comment by drcrallen** [comment]
Other than what @cheddar mentioned, this looks good to me. I can do the merge once the conflicts are cleaned up.

I did some more digging on the `<p>` thing and I think I had it backwards. This is in the master pom: https://github.com/druid-io/druid/blob/druid-0.9.2-rc2/pom.xml#L787 which I *think* prevents the problem I was describing.

I think my brain was backwards, where bytebuffer-collections would fail compiling if java8 was used.

Looking through druid source code there are a few instances of `<p>` which should have otherwise given us compile problems.

So you can fix the merge conflicts and I'll merge this promptly assuming another committer can give the other plus one.

**Comment by akashdw** [comment]
@cheddar @drcrallen @fjy I uploaded a new patch set to resolve code conflicts and removed <p> from bytebuffer-collections.

**Comment by fjy** [comment]
👍

**Comment by leventov** [comment]
@akashdw could you also please remove `bytebuffer-collections` dependency from the root pom?

**Comment by akashdw** [comment]
@leventov Sure, I will create a pr for the pom cleanup.

**Comment by akashdw** [comment]
@leventov pr for root pom cleanup, https://github.com/druid-io/druid/pull/3764

**Pull request merged**